Cubical/Categories/Instances/PosetCat/Monotone.agda

```diff
@@ -0,0 +1,207 @@
+-- both preorders and posets.
+
+module Cubical.Categories.Instances.PosetCat.Monotone where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Function hiding (_$_)
+open import Cubical.Foundations.Structure
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Relation.Binary.Base
+open import Cubical.Reflection.Base
+open import Cubical.Reflection.RecordEquiv
+open import Cubical.Data.Sigma
+
+open import Cubical.Relation.Binary.Poset
+
+open BinaryRelation
+
+
+private
+  variable
+    ℓ ℓ' : Level
+
+
+
+-- If Y is a set, then functions into Y form a set
+isSetArrow : {X Y : Type} -> isSet Y -> isSet (X -> Y)
+isSetArrow Hy = λ f g p q i j x →
+  Hy (f x) (g x) (λ k -> p k x) (λ k -> q k x) i j
+
+
+module _ {ℓ ℓ' : Level} where
+
+  -- Because of a bug with Cubical Agda's reflection, we need to declare
+  -- a separate version of MonFun where the arguments to the isMon
+  -- constructor are explicit.
+  -- See https://github.com/agda/cubical/issues/995.
+  module _ {X Y : Poset ℓ ℓ'} where
+
+    module X = PosetStr (X .snd)
+    module Y = PosetStr (Y .snd)
+    _≤X_ = X._≤_
+    _≤Y_ = Y._≤_
+
+    monotone' : (⟨ X ⟩ -> ⟨ Y ⟩) -> Type (ℓ-max ℓ ℓ')
+    monotone' f = (x y : ⟨ X ⟩) -> x ≤X y → f x ≤Y f y       
+
+    monotone : (⟨ X ⟩ -> ⟨ Y ⟩) -> Type (ℓ-max ℓ ℓ')
+    monotone f = {x y : ⟨ X ⟩} -> x ≤X y → f x ≤Y f y   
+
+  -- Monotone functions from X to Y
+  -- This definition works with Cubical Agda's reflection.
+  record MonFun' (X Y : Poset ℓ ℓ') : Type ((ℓ-max ℓ ℓ')) where
+    field
+      f : (X .fst) → (Y .fst)
+      isMon : monotone' {X} {Y} f
+
+  -- This is the definition we want, where the first two arguments to isMon
+  -- are implicit.
+  record MonFun (X Y : Poset ℓ ℓ') : Type ((ℓ-max ℓ ℓ')) where
+    field
+      f : (X .fst) → (Y .fst)
+      isMon : monotone {X} {Y} f
+
+  open MonFun
+
+  isoMonFunMonFun' : {X Y : Poset ℓ ℓ'} -> Iso (MonFun X Y) (MonFun' X Y)
+  isoMonFunMonFun' = iso
+    (λ g → record { f = MonFun.f g ; isMon = λ x y x≤y → isMon g x≤y })
+    (λ g → record { f = MonFun'.f g ; isMon = λ {x} {y} x≤y -> MonFun'.isMon g x y x≤y })
+    (λ g → refl)
+    (λ g → refl)
+
+
+  isPropIsMon : {X Y : Poset ℓ ℓ'} -> (f : MonFun X Y) ->
+    isProp (monotone {X} {Y} (MonFun.f f))
+  isPropIsMon {X} {Y} f =
+    isPropImplicitΠ2 (λ x y ->
+      isPropΠ (λ x≤y -> IsPoset.is-prop-valued (PosetStr.isPoset (str Y)) (MonFun.f f x) (MonFun.f f y)))
+
+  isPropIsMon' : {X Y : Poset ℓ ℓ'} -> (f : ⟨ X ⟩ -> ⟨ Y ⟩) ->
+    isProp (monotone' {X} {Y} f)
+  isPropIsMon' {X} {Y} f =
+    isPropΠ3 (λ x y x≤y -> IsPoset.is-prop-valued (PosetStr.isPoset (str Y)) (f x) (f y))
+
+
+ -- Equivalence between MonFun' record and a sigma type   
+  unquoteDecl MonFun'IsoΣ = declareRecordIsoΣ MonFun'IsoΣ (quote (MonFun'))
+
+  Sigma : (X Y : Poset ℓ ℓ') -> Type (ℓ-max ℓ ℓ')
```

Comment:
What is this for?

Reply:
I'm pretty sure that it's to define the isomorphism to a sigma type to get the `isSet` property nicely, but not really sure -- stole this file from the other repo.